strawberry/exceptions.py

```diff
@@ -14,6 +14,13 @@ def __init__(self):
         super().__init__(message)
 
 
+class NotAClass(Exception):
+    def __init__(self, method: str):
+        message = f"strawberry.{method} can only be used with classes"
+
+        super().__init__(message)
```

Comment:
This should just take the object itself, and not the `str` (i.e. the exception should be in charge of the output formatting)

Reply:
Mmm but if I give the `cls` object to the exception, this will be the non-class object, I need to pass the method the user is using: `type`|`input`|`interface`... 🤔

Reply:
Oh right, I misread how this error was to be used. Hmmm, in that case I'm not a fan of passing the strings directly. Maybe we can use some `classmethod`s here?

```python
from __future__ import annotations
from enum import Enum

class NotAClass(Exception):
    class MethodType(Enum):
        INPUT = "input"
        INTERFACE = "interface"
        TYPE = "type"

    def __init__(self, obj: object, method_type: METHOD_TYPE):
        message = (
            f"strawberry.{method_type} can only be used with class types. Provided object "
            f"{obj} is not a type."
        )

        super().__init__(message)

   @classmethod
   def input(cls, obj: object) -> NotAClass:
       return cls(obj, cls.MethodType.INPUT)

   @classmethod
   def interface(cls) -> NotAClass:
       return cls(obj, cls.MethodType.INTERFACE)

   @classmethod
   def type(cls) -> NotAClass:
       return cls(obj, cls.MethodType.TYPE)
```
```python
a = object()  # example object that's not a type

raise NotAClass.interface(a)
```

Reply:
Nice! much better :)